components/admin_console/permission_schemes_settings/guest_permissions_tree/guest_permissions_tree.tsx

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import Permissions from 'mattermost-redux/constants/permissions';
+import {Role} from 'mattermost-redux/types/roles';
+import {ClientLicense} from 'mattermost-redux/types/config';
+
+import PermissionGroup from '../permission_group.jsx';
+import EditPostTimeLimitButton from '../edit_post_time_limit_button';
+import EditPostTimeLimitModal from '../edit_post_time_limit_modal';
+
+type Props = {
+    license: ClientLicense;
+    onToggle: (role: string, permissions: string[]) => void;
+    readOnly: boolean;
+    scope: string;
+    selectRow: (permission: string) => void;
+    parentRole?: Role;
+    selected?: string | null;
+    role?: Partial<Role>;
+}
+
+const GuestPermissionsTree = ({license, onToggle, readOnly, scope, selectRow, parentRole, selected, role = {permissions: []}}: Props) => {
+    const [editTimeLimitModalIsVisible, setEditTimeLimitModalIsVisible] = React.useState(false);
+
+    const openPostTimeLimitModal = () => setEditTimeLimitModalIsVisible(true);
+    const closePostTimeLimitModal = () => setEditTimeLimitModalIsVisible(false);
+    const toggleGroup = (ids: string[]) => {
+        if (readOnly) {
+            return;
+        }
+        onToggle(role.name!, ids);
+    };
+
+    const ADDITIONAL_VALUES = {
```

Comment:
Similarly, could you wrap `ADDITIONAL_VALUES` and `permissions` in a `useMemo` for roughly the same reason?

Reply:
Another good point for variables :)